presto-testing/src/main/java/io/prestosql/testing/CountingMockConnector.java

```diff
@@ -101,6 +121,19 @@ private ConnectorFactory getConnectorFactory()
                     getColumnsCallsCounter.incrementAndGet();
                     return defaultGetColumns().apply(schemaTableName);
                 })
+                .withRoleGrants((connectorSession, roles, grantees, limit) -> {
+                    listRowGrantsCallsCounter.incrementAndGet();
+                    if (roles.isPresent()) {
+                        rolesPushedCounter.incrementAndGet();
+                    }
+                    if (grantees.isPresent()) {
+                        granteesPushedCounter.incrementAndGet();
+                    }
+                    if (limit.isPresent()) {
+                        limitPushedCounter.incrementAndGet();
```

Comment:
`listRowGrantsCallsCounter`, `rolesPushedCounter`, `rolesPushedCounter` and `limitPushedCounter` should be grouped into a single bean that represents interaction with metadata call. Typically we had single counter per method, you had 4. Using bean we would maintain having single counter per method, where one counter is just complex.

Reply:
I added beans: a Counter bean a Count bean to compare against.